### ISSUE TITLE
Feature request: Exact age

### ISSUE THREAD

**Issue by mvsmal** [date-fns/date-fns#840]
I one of my projects I needed to show the exact age of a person, in a format: 22 years 6 months 27 days.
Unfortunately I didn't find any suitable function in date-fns library.
I found several algorithms in internet doing this calculation, but none of them are human-friendly, basically they only count years as 365 days and months as 30 days, which is not exactly what I wanted.

So I implemented my own algorithm and I would like to share it as a part of this library. I only need to agree on the name of such function and also get some support for internalization of it.

My proposals:
1. Return a string. Depending on options output/not output zeros, and output/not output hours, minutes. This will need internalization.
```javascript
function humanAge(date, options) {
 //code
 return 'Y years M months D days';
}
```

2. Return just an object and a user would handle the numbers by himself. Not sure about hours and minutes. This one doesn't need internalization.
```javascript
function humanAge(date) {
  // code
  return {
    years: Y,
    months: M,
    days: D,
    hours: H, // ?
    minutes: m // ?
  }
}
```

Basically I can implement both functions

**Comment by pbrianmackey** [comment]
I downloaded this library to solve this problem.  Now I see it isn't supported.

**Comment by omonk** [comment]
@mvsmal - can you share your own solution in the issue? I appreciate its been quite a while but would be good to share for others who come across this issue! 😄

**Comment by davidmillercode** [comment]
Would also really appreciate this feature.

We are currently taking a datestring and `differenceInYears` is not accurate depending on the datestring format.  A function to create a date from a datestring would also suffice.

**Comment by delkawy** [comment]
My solution using date-fns

```
function formatIdade(date) {
  const result = [];
  const now = new Date();
  let age = parseISO(date);

  const years = differenceInYears(now, age);
  if (years > 0) {
    result.push(`${years} years`);
    age = addYears(age, years);
  }

  const months = differenceInMonths(now, age);
  if (months > 0) {
    result.push(`${months} months`);
    age = addMonths(age, months);
  }

  const days = differenceInDays(now, age);
  if (days > 0) {
    result.push(`${days} days`);
  }

  return result.join(', ');
}
```

**Comment by evdama** [comment]
> In one of my projects I needed to show the exact age of a person, in a format: 22 years 6 months 27 days.

Exactly what I need as well 😃
What's the status on this issue/feature?

**Comment by evdama** [comment]
Is there any news regarding this i.e. does anyone know, based on discussions maybe, whether or not this is considered to be added to date-fns or not?

**Comment by kkor** [comment]
Doesn't https://date-fns.org/v2.14.0/docs/intervalToDuration solve this problem already?

**Comment by toinuma** [comment]
yes
The `years` I could get from `intervalToDuration` was synonymous with age.
Thank you very much!

**Comment by calvintennant** [comment]
This is already supported.

```js
formatDuration(intervalToDuration({start: parseISO('1994-09-22T00:00:00'), end: new Date()}), {format: ['years', 'months', 'days']})
```

results in `28 years 21 days`.

Should probably close this issue.
